Terraform.sublime-syntax

```diff
@@ -699,50 +717,103 @@ contexts:
   # Blocks: Identifier (StringLit|Identifier)* "{" Newline Body "}" Newline;
   #
   # https://github.com/hashicorp/hcl2/blob/master/hcl/hclsyntax/spec.md#structural-elements
-  block:
-    # Special case heuristic for the AWS and friends two-term resources
-    - match: (\b(resource)\s+(")({{identifier}})(")\s+(")({{identifier}})("))\s*(\{)
-      captures:
-        1: meta.type.terraform
-        2: keyword.declaration.terraform
-        3: punctuation.definition.begin.terraform
-        4: support.type.terraform
-        5: punctuation.definition.end.terraform
-        6: punctuation.definition.begin.terraform
-        7: entity.name.type.terraform
-        8: punctuation.definition.end.terraform
-        9: meta.block.terraform punctuation.section.block.begin.terraform
-      push: block-body
-    # Generic
-    - match: (?:\b({{terraform_known_blocks}})\b|({{identifier}}))(?=[-\s\w"]*\{)
-      scope: meta.type.terraform
-      captures:
-        1: keyword.declaration.terraform
-        2: entity.name.type.terraform
-      push: block-name
+  blocks:
+    - include: typed-and-named-blocks
+    - include: named-blocks
+    - include: generic-blocks
+
+  typed-and-named-blocks:
+    # Special case for two-label resources,
+    # where the first denotes the type and the second the name.
+    - match: (?=\b{{terraform_typed_and_named_blocks}}(\s+{{label}}){2}\s*\{)
+      push:
+        - body
+        - block-name-label
+        - block-type-label
+        - block-declaration
+
+  named-blocks:
+    # Special case for two-label resources,
+    # where the first denotes the type and the second the name.
+    - match: (?=\b{{terraform_named_blocks}}\s+{{label}}\s*\{)
+      push:
+        - body
+        - block-name-label
+        - block-declaration
 
-  block-name:
-    - meta_content_scope: meta.type.terraform
+  generic-blocks:
+    - match: (?=\b({{identifier}})(\s+{{label}})*\s*\{)
+      push:
+        - body
+        - generic-block-labels
+        - block-declaration
+
+  block-declaration:
+    - match: \b{{identifier}}\b
```

Comment:
I've probably directly consumed declaration keywords followed by lookahead for amount of labels instead of matching initial identifier twice to save some roundtrips, but it probably doesn't make a big difference.

As leading `\b` is already matched and lookaheads only appliy if keyword is followed by whitespace or brace, following should be sufficient.

```suggestion
    - match: '{{identifier}}'
```